native/src/rapidsml_jni.cu

```diff
@@ -247,16 +269,64 @@ JNIEXPORT void JNICALL Java_com_nvidia_spark_ml_linalg_JniRAPIDSML_dgemm(JNIEnv*
     env->ThrowNew(jlexception, "Error freeing C from device");
   }
 
-  status = cublasDestroy(handle);
-  if (status != CUBLAS_STATUS_SUCCESS) {
-    env->ThrowNew(jlexception, "Error destroying cuBLAS handle");
-  }
-
   env->ReleaseDoubleArrayElements(A, host_A, JNI_ABORT);
   env->ReleaseDoubleArrayElements(B, host_B, JNI_ABORT);
   env->ReleaseDoubleArrayElements(C, host_C, 0);
 }
 
+JNIEXPORT jlong Java_com_nvidia_spark_ml_linalg_JniRAPIDSML_dgemmWithColumnViewPointer(JNIEnv *env, jclass, jint transa, jint transb, jint m, jint n,
+                                                                                      jint k, jdouble alpha, jdoubleArray A, jint lda, jlong B,
+                                                                                      jint ldb, jdouble beta, jint ldc, jint deviceID)
+{
+  cudaSetDevice(deviceID);
+  raft::handle_t raft_handle;
+  cudaStream_t stream = raft_handle.get_stream();
+
+  try {
+    cudf::column_view *B_cv_ptr = reinterpret_cast<cudf::column_view *>(B);
+    // init cuda stream view from rmm
+    auto c_stream = rmm::cuda_stream_view(reinterpret_cast<cudaStream_t>(stream));
+
+    auto size_A = env->GetArrayLength(A);
+    auto *host_A = env->GetDoubleArrayElements(A, nullptr);
+    rmm::device_buffer dev_buff_A = rmm::device_buffer(host_A, size_A * sizeof(double), c_stream);
+    auto size_C = m * n;
+    //create child column that will own the computation result
+    auto child_column = cudf::make_numeric_column(cudf::data_type{cudf::type_id::FLOAT64}, size_C);
+    auto child_mutable_view = child_column->mutable_view();
+    auto status = raft::linalg::cublasgemm(raft_handle.get_cublas_handle(), convertToCublasOpEnum(transa), convertToCublasOpEnum(transb),
+                                           m, n, k, &alpha, (double *)dev_buff_A.data(), lda, (double *)B_cv_ptr->data<double>(),
+                                           ldb, &beta, (double *)child_mutable_view.data<double>(), ldc, stream);
```

Comment:
The casts on the column view `data` calls should not be necessary since it's already templated on the type.
```suggestion
                                           m, n, k, &alpha, (double const *)dev_buff_A.data(), lda, B_cv_ptr->data<double>(),
                                           ldb, &beta, child_mutable_view.data<double>(), ldc, stream);
```

Reply:
Added.